### PR TITLE
Advanced commads

### PR DESCRIPTION
**Pull request by jcgertig**
@nitin42 this gives the users more options in the way commands are used

### CODE REVIEW

**Comment by jcgertig** [comment]
Example code in the readme and usage like so
example

**Comment by jcgertig** [comment]
also fixed the scroll issue #11

**Comment by jcgertig** [comment]
Added a `runCommand` method so commands can run other commands
an example
```javascript
'type-text': (args, print, runCommand) => {
  const text = args.slice(1).join(' ');
  print('');
  for (let i = 0; i < text.length; i += 1) {
    setTimeout(() => {
      runCommand(`edit-line ${text.slice(0, i + 1)}`);
    }, 100 * i);
  }
},
```

cool stuff bro

**Comment by nitin42** [comment]
> Added a `runCommand` method so commands can run other commands
an example

SUPERB!!!! Looks really great!

Great job 👍

**Comment by nitin42** [comment]
Seems fine to me! But before I merge this to master, have you tested everything ? Every feature ? from your side ?

File: README.md [hunk 1]
```diff
@@ -98,13 +98,47 @@ You can have the terminal pass out the cmd that was input
...
-  commandPassThrough={(cmd, done) => {
+  commandPassThrough={(cmd, print) => {
     // do something async
-    done(`-PassedThrough:${cmd}: command not found`);
+    print(`-PassedThrough:${cmd}: command not found`);
   }}
 />
 ```
 
+## Advanced commands
```
Comment (nitin42) [inline README.md]: Thanks for adding the docs also 😄

File: components/Content.js [hunk 2]
```diff
@@ -40,15 +44,15 @@ class Content extends Component {
         className="terminal-container terminal-container-main"
         style={(
           backgroundColor,
-          maximise ? { maxWidth: '100%' } : null
+          maximise ? { maxWidth: '100%', maxHeight: 'calc(100% - 30px)' } : null
```
Comment (nitin42) [inline components/Content.js]: 👍

File: components/Terminal.js [hunk 3]
```diff
@@ -118,21 +129,17 @@ class Terminal extends Component {
...
   setDescription = () => {
-    this.setState({
-      description: {
-        show: 'show the msg',
-        ...this.props.description,
-        clear: 'clear the screen',
-        help: 'list all the commands',
-      },
-    });
+    const description = {
```
Comment (nitin42) [inline components/Terminal.js]: Thanks for refactoring this!

File: components/Terminal.js [hunk 4]
```diff
@@ -222,6 +262,30 @@ class Terminal extends Component {
     }
   };
 
+  runCommand = (inputText) => {
```
Comment (nitin42) [inline components/Terminal.js]: Awesome ! It's really a cool feature.

**Comment by nitin42** [comment]
I must say that I'm really enjoying this stuff more now. Thanks for adding this awesome features !!

**Comment by nitin42** [comment]
Btwn @jcgertig we should add more examples covering this ? What you think ?

**Comment by nitin42** [comment]
@jcgertig see this



Not working as expected! Do you know why ?

**Pull request merged**